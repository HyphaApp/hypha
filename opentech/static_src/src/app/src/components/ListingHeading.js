import React from 'react';
import PropTypes from 'prop-types';

export default class ListingHeading extends React.Component {
    render() {
        return (
            <li className="listing__item listing__item--heading">
                <h5 className="listing__title">{this.props.title}</h5>
                <span className="listing__count">{this.props.count}</span>
            </li>
        );
    }
}

ListingHeading.propTypes = {
    title: PropTypes.string,
    count: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
    ]),
};
