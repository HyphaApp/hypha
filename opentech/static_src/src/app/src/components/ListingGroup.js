import React from 'react';
import PropTypes from 'prop-types';

import ListingHeading from '@components/ListingHeading';

export default class ListingGroup extends React.Component {
    render() {
        return (
            <>
                <ListingHeading title={this.props.item.group} count={this.props.children.length} />
                <ul>
                    {this.props.children}
                </ul>
            </>
        );
    }
}

ListingGroup.propTypes = {
    item: PropTypes.shape({
        group: PropTypes.string,
    }),
};
