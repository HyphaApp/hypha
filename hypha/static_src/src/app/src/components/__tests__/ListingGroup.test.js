import React from "react";
import { mount } from "enzyme";
import sinon from "sinon";
import ListingGroup from "../ListingGroup";
import ListingHeading from "../ListingHeading";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });

describe("Test Listing Group component", () => {
    const item = {name : "title text"}
    const id = "id"
    const children = [<span key={1}>Children</span>]
    const subject = mount(<ListingGroup item={item} id={id} children={children}>
      </ListingGroup>
    );
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and elements", () => {
      expect(subject.find('li').props().id).toBe(id);
      expect(subject.containsMatchingElement(<ListingHeading />)).toEqual(true);

      expect(subject.find('ListingHeading').first().props().title).toBe(item.name);
      expect(subject.find('ListingHeading').first().props().count).toBe(1);

      expect(subject.find('ul').length).toBe(1);
    });
  
    test('render a listing group component', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });
