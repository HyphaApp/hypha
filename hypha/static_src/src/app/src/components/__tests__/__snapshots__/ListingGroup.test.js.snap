// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Listing Group component render a listing group component 1`] = `
<ListingGroup
  id="id"
  item={
    Object {
      "name": "title text",
    }
  }
>
  <li
    id="id"
  >
    <ListingHeading
      count={1}
      title="title text"
    >
      <div
        className="listing__item listing__item--heading"
        id="title-text"
      >
        <h5
          className="listing__title"
        >
          title text
        </h5>
        <span
          className="listing__count"
        >
          1
        </span>
      </div>
    </ListingHeading>
    <ul>
      <span
        key="1"
      >
        Children
      </span>
    </ul>
  </li>
</ListingGroup>
`;
